### ISSUE TITLE
small object detection

### ISSUE THREAD

**Issue by ssetty** [fizyr/keras-retinanet#816]
I want to detect small objects like cracks on a big metal surface. Size of object to be detected is very small compared to overall image size.  What parameters do I need to change?
I see AnchorParameters in anchor.py - how do I change these values to fit small objects?

In yolo we have generateAnchors which uses knn to estimate anchor size - anything similar for rcnn?

Thanks

**Comment by ssetty** [comment]
For small object detection I changed anchor parameters results are very bad - no detection.

AnchorParameters.default = AnchorParameters(
    sizes  = [8,16,32,64,128],
    strides=[2,4,8,16,32],
    ratios  = np.array([0.5, 1, 2], keras.backend.floatx()),
    scales  = np.array([2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)], keras.backend.floatx()),
) 

Please help.

**Comment by hgaiser** [comment]
You'll mainly want to experiment with the ratios and scales values to see if you can get better coverage on your objects. Try out some values and inspect the coverage with `debug.py`. This is more or less what YOLO does too, but automated. If you implement such a nice feature in keras-retinanet, a PR would be welcome.

For now, I'll close this issue. For further discussions I'd suggest trying the #keras-retinanet slack channel.

**Comment by mazatov** [comment]
Hey @ssetty , any luck with small objects? I'm dealing with it now.

**Comment by ssetty** [comment]
Hi - it is becoming worse after changing anchor parameters. Thanks

**Comment by Sarvesh-Angadi** [comment]
hey,I am dealing with the same problem?Any solutions?

**Comment by mazatov** [comment]
Not really. I managed to change the values of anchors to get most of my labels annotated by debug.py, but the performance is still not good. I'm getting much better results with YOLO3.

**Comment by Sarvesh-Angadi** [comment]
Thanks!
